src/pages/_app.js

```diff
@@ -12,9 +13,11 @@ function MyApp({ Component, pageProps }) {
     }, []);
     return (
         <ThemeProvider theme={theme}>
-            <CyverseAppBar>
-                <Component {...pageProps} />
-            </CyverseAppBar>
+            <UploadTrackingProvider>
+                <CyverseAppBar>
+                    <Component {...pageProps} />
+                </CyverseAppBar>
+            </UploadTrackingProvider>
         </ThemeProvider>
```

Comment:
This may be my ignorance. Do we really need to wrap the whole app with the `Provider`. Is it not enough to just wrap the components in `data` page ? In other words, are we going to display the queue at the app level or just in the data page ?

Reply:
My understanding is that they need to be wrapped in the same instance of UploadTrackingProvider. Could be wrong, though. My plan is to hopefully be able to move that tag closer to where it's used later, once we're iterating less quickly on the file listing.

A wrinkle will be the fact that the upload queue component will be viewable from basically anywhere in the app. That will make finding the closest common ancestor node a bit of a challenge.